google/cloud/aiplatform/training_jobs.py

```diff
@@ -281,46 +333,474 @@ def package_and_copy_to_gcs(
+                training script.
+            model_serving_container_predict_route (str):.
+                If the training produces a managed AI Platform Model, An HTTP path to
+                send prediction requests to the container, and which must be supported
+                by it. If not specified a default HTTP path will be used by AI Platform.
+            model_serving_container_health_route (str):
+                If the training produces a managed AI Platform Model, an HTTP path to
+                send health check requests to the container, and which must be supported
+                by it. If not specified a standard HTTP path will be used by AI
+                Platform.
+            project (str):
+                Project to run training in. Overrides project set in aiplatform.init.
+            location (str):
+                Location to run training in. Overrides location set in aiplatform.init.
+            credentials (auth_credentials.Credentials):
+                Custom credentials to use to run call training service. Overrides
+                credentials set in aiplatform.init.
+            staging_bucket (str):
+                Bucket used to stage source and training artifacts. Overrides
+                staging_bucket set in aiplatform.init.
+        """
+        super().__init__(project=project, location=location, credentials=credentials)
+        self._display_name = display_name
+        self._script_path = script_path
+        self._container_uri = container_uri
+        self._requirements = requirements
+        self._staging_bucket = staging_bucket
+        self._project = project
+        self._credentials = credentials
+        self._model_serving_container_image_uri = model_serving_container_image_uri
+        self._model_serving_container_predict_route = (
+            model_serving_container_predict_route
+        )
+        self._model_serving_container_health_route = (
+            model_serving_container_health_route
+        )
+        self._gca_resource = None
+
+    # TODO(b/172365904) add filter split, training_pipline.FilterSplit
+    # TODO(b/172366411) predefined filter split training_pipeline.PredfinedFilterSplit
+    # TODO(b/172368070) add timestamp split, training_pipeline.TimestampSplit
     def run(
         self,
-        dataset: Optional[datasets.Dataset],
-        output_dir: Optional[str] = None,
+        dataset: Optional[datasets.Dataset] = None,
+        model_display_name: Optional[str] = None,
+        base_output_dir: Optional[str] = None,
         args: Optional[Dict[str, str]] = None,
-        replica_count=0,
-        machine_type="n1-standard-4",
-        accelerator_type=None,
-        accelerator_count=0,
-        spec_yaml_path: Optional[str] = None,
-    ) -> models.Model:
-        pass
+        replica_count: str = 0,
+        machine_type: str = "n1-standard-4",
+        accelerator_type: str = "ACCELERATOR_TYPE_UNSPECIFIED",
```

Comment:
This is fine, but perhaps consider directly using the existing GAPIC Enums for this?

Reply:
I went with this because we don't seem to have an enum for machine type so it seems inconsistent to require a string for one hardware type and an enum for another. Though, as we receive feedback we can switch.